### PR TITLE
Make fuzzy zkapp test bounded by a max timeout

### LINKED ISSUE #13281: Changes for fuzzy test
The changes to make in the tests are:

1. Update the tests to use a non-deterministic seed. I don't know how non-determinstic the system entropy will be on a docker container (if someone knows, please inform); we could try and sample randomness from the internet instead, but anything will be an improvement for the first pass.
    Rather than taking in a static number of trials and then running each test case for that many trials, we should configure the tests with a "maximum timeout". It would then make sense to basically run one test of each case in a loop until that timeout is met. We can configure the timeout via the CLI so that it's easy to configure the timeout in the buildkite configuration. Once the timeout is hit, we will stop executing further tests (and possibly cancel the currently running test if each individual test takes too long).
 2. I'm not sure if this is already the case or not, but we shouldn't crash when we hit a test failure. Instead, we should print all errors we encounter when testing, and keep fuzzing until the timeout. The status code we exit with should be 0 if no test failures were encountered, and 1 if any test failures were encountered.

### PR DESCRIPTION
**Pull request by ghost-not-in-the-shell**
Explain your changes:
This PR changed the `zkapp_fuzzy` test to be time-bounded. Now it takes a command line argument of `timeout` that configures how long the test would run. And during that time, the test would iteratively execute each individual test.

Explain how you tested your changes:
*

Checklist:

- [ ] Dependency versions are unchanged
  - Notify Velocity team if dependencies must change in CI
- [ ] Modified the current draft of release notes with details on what is completed or incomplete within this project
- [ ] Document code purpose, how to use it
  - Mention expected invariants, implicit constraints
- [ ] Tests were added for the new behavior
  - Document test purpose, significance of failures
  - Test names should reflect their purpose
- [ ] All tests pass (CI will check this if you didn't)
- [ ] Serialized types are in stable-versioned modules
- [ ] Does this close issues? List them

* Closes #13281

### CODE REVIEW

**Comment by ghost-not-in-the-shell** [comment]
!ci-build-me

File: src/lib/transaction_snark/test/zkapp_fuzzy/zkapp_fuzzy.ml [hunk 1]
```diff
@@ -91,177 +92,285 @@ let mk_ledgers_and_fee_payers ?(is_timed = false) ~num_of_fee_payers () =
...
+        zkapp_command_dummy_auths
+    in
+    let%map () =
+      match%map
+        try_with (fun () ->
+            with_timeout (Time.Span.of_int_sec individual_test_timeout)
+            @@ U.check_zkapp_command_with_merges_exn ~global_slot ledger
+                 [ zkapp_command ] )
+      with
+      | Ok (`Result ()) ->
```
Comment (mrmr1993) [inline src/lib/transaction_snark/test/zkapp_fuzzy/zkapp_fuzzy.ml:121]: Nit: this match statement boilerplate is repeated a few times. Would be nicer to factor it out as a function.

File: buildkite/scripts/fuzzy-zkapp-test.sh [hunk 2]
```diff
@@ -9,7 +9,8 @@ fi
 
 profile=$1
 path=$2
-trials=$3
+timeout=$3
+individual_test_timeout=$4
```
Comment (mrmr1993) [inline buildkite/scripts/fuzzy-zkapp-test.sh:13]: You need to change the `3` on line 5 to a `4` for this to work. Usage description should also change.

**Review by dkijania — APPROVED** [review]

File: src/lib/transaction_snark/test/zkapp_fuzzy/zkapp_fuzzy.ml (hunk 1 above)
Comment (ghost-not-in-the-shell) [reply src/lib/transaction_snark/test/zkapp_fuzzy/zkapp_fuzzy.ml:121]: The match statement has some difference in the successful and failure cases. If we want to abstract this out, we also need to include the generation of the zkapp. I would leave it as it is for now.

**Review by shimkiv — APPROVED** [review]

**Comment by ghost-not-in-the-shell** [comment]
!ci-build-me

**Review by mrmr1993 — APPROVED** [review]

**Comment by ghost-not-in-the-shell** [comment]
!ci-build-me

**Comment by deepthiskumar** [comment]
!approved-for-mainnet

**Pull request merged**